### PR TITLE
Updates doc about internal memory management

### PR DESCRIPTION
**Pull request by cangermueller**
Updates documentation about internal Python memory management as described in #2652:
* Adds `list(...)` in code example to get similar results with python3
* Add note that results may different for different python versions and platforms
* Rewrites section to make it more clear

### CODE REVIEW

File: doc/tutorial/python-memory-management.txt [hunk 1]
```diff
@@ -307,7 +307,7 @@ invoking
 
 ::
 
-    python -m memory_profiler memory-profile-me.py
+    python2.7 -m memory_profiler memory-profile-me.py
```
Comment (abergeron) [inline doc/tutorial/python-memory-management.txt]: I disagree with this change.

**Comment by abergeron** [comment]
Except for that one part, I think this is ok.  I'm not 100% sure about the rewrite of the explanation paragraphs, but it doesn't look much worse than what was there before.

**Comment by cangermueller** [comment]
@abergeron, I agree with you: the python version is not so relevant. I changed `python2.7` to `python`

**Comment by abergeron** [comment]
I just realized that you might want to update the section with the output of memory_profile since it does not show the same code as the code listing you changed.

**Comment by cangermueller** [comment]
Done!

**Pull request merged**